gokitmiddlewares/stalemiddleware/middleware.go

```diff
@@ -0,0 +1,39 @@
+package stalemiddleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/arquivei/foundationkit/app"
+	"github.com/go-kit/kit/endpoint"
+)
+
+func New(c Config) endpoint.Middleware {
+	ch := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ch:
+				if !app.IsHealthy() {
+					app.SetHealthy()
+				}
+			case <-time.After(c.Timeout):
+				if app.IsHealthy() {
+					if c.Logger != nil {
+						c.Logger.Warn().
+							Dur("timeout", c.Timeout).
+							Msg("Endpoint didn't receive any request and it's stale")
```

Comment:
Nit: this function is starting to become a hadouken code. I imagine breaking it into this functions:

* `backgroundStaleCheck()` // everything inside `go func() { ...`
* `setUnhealthy()` // everything inside time.After